File: Math-Array-String/10.py
# 2427. Number of Common Factors

class Solution:
    def commonFactors(self, a: int, b: int) -> int:
        count = 0
        for i in range(1, min(a, b) + 1):
            if not a % i and not b % i :
            # if a % i == 0 and b % i == 0:
                count += 1
        return count
    

sol = Solution()
output = sol.commonFactors(25, 30)
print(output)

File: JavaScript/Math-Array-String/10.js
// 2427. Number of Common Factors

/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var commonFactors = function(a, b) {

    let mi = Math.min(a,b);
    let count = 0
    for (let i = 1; i <= mi; i++) {
        if (a % i === 0 && b % i === 0) {
            count++;
        }
    }
    return count;
    
};

console.log(commonFactors(25, 30));